src/remodel_api/remodel.rs

```diff
@@ -325,10 +377,87 @@ impl Remodel {
         rbx_binary::to_writer_default(&mut buffer, &tree, instance.children())
             .map_err(rlua::Error::external)?;
 
-        Remodel::upload_asset(context, buffer, asset_id)
+        Remodel::upload_asset(context, buffer, asset_id, queries)
+    }
+
+    fn write_new_model_asset(
+        context: Context<'_>,
+        lua_instance: LuaInstance,
+        name: String,
+        description: String,
+        is_public: bool,
+        allow_comments: bool,
+    ) -> rlua::Result<String> {
+        let is_public = bool_into_query(is_public);
+        let allow_comments = bool_into_query(allow_comments);
+
+        Remodel::write_model_asset(
+            context,
+            lua_instance,
+            0,
+            &[
+                ("type", "Model"),
+                ("name", &name),
+                ("description", &description),
+                ("isPublic", &is_public),
+                ("allowComments", &allow_comments),
+            ],
+        )
+    }
+
+    fn write_new_place_asset(
+        context: Context<'_>,
+        lua_instance: LuaInstance,
+        name: String,
+        description: String,
+        is_public: bool,
+        allow_comments: bool,
+    ) -> rlua::Result<String> {
+        let is_public = bool_into_query(is_public);
+        let allow_comments = bool_into_query(allow_comments);
+
+        Remodel::write_place_asset(
+            context,
+            lua_instance,
+            0,
+            &[
+                ("type", "Model"),
+                ("name", &name),
+                ("description", &description),
+                ("isPublic", &is_public),
+                ("allowComments", &allow_comments),
+            ],
+        )
+    }
+
+    fn write_existing_model_asset(
+        context: Context<'_>,
+        lua_instance: LuaInstance,
+        asset_id: u64,
+    ) -> rlua::Result<()> {
+        match Remodel::write_model_asset(context, lua_instance, asset_id, &[]) {
+            Ok(_) => Ok(()),
+            Err(error) => Err(rlua::Error::external(error)),
+        }
     }
 
-    fn upload_asset(context: Context<'_>, buffer: Vec<u8>, asset_id: u64) -> rlua::Result<()> {
+    fn write_existing_place_asset(
+        context: Context<'_>,
+        lua_instance: LuaInstance,
+        asset_id: u64,
+    ) -> rlua::Result<()> {
+        match Remodel::write_place_asset(context, lua_instance, asset_id, &[]) {
+            Ok(_) => Ok(()),
+            Err(error) => Err(rlua::Error::external(error)),
+        }
```

Comment:
Same here as in the other case. Matching here doesn't really make sense when we have `?`.